ts/sagas/pinlogin.ts

```diff
@@ -30,22 +47,34 @@ function* pinValidateSaga(action: PinValidateRequest): Iterator<Effect> {
   try {
     const userPin = action.payload;
     const basePin: string = yield call(getPin);
+
     if (basePin === userPin) {
-      // Navigate to the MainNavigator
-      const navigateToPinValidNavigatorAction = NavigationActions.navigate({
-        routeName: ROUTES.MAIN,
-        key: undefined
-      });
-      yield put(navigateToPinValidNavigatorAction);
+      const deeplink: NavigationNavigateActionPayload | null = yield select(
+        deeplinkSelector
+      );
+      // tslint:disable-next-line:readonly-array array-type
+      const actionsToDispatch: Array<PutEffect<Action>> = [put(pinSuccess())];
+
+      // If a deeplink has been setted, navigate to deeplink, otherwise to the MainNavigator
+      if (deeplink) {
+        actionsToDispatch.push(put(navigateToDeeplink(deeplink)));
```

Comment:
I remember I already reviewed this logic, I would prefer to avoid the mutable array, shouldn't the following be equivalent?

```ts
yield put(pinSuccess());
if(deeplink) {
  yield put(navigateToDeeplink(deeplink))
} else {
  yield put(
     NavigationActions.navigate({
         routeName: ROUTES.MAIN,
         key: undefined
      })
   ) ;
}

Reply:
Refactored.